### PR TITLE
build: set SENPERFORMANCE=2 for the perf benchmark suite

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
The `TEST_TYPE=perf` target shells out to `spyre-perf-suite`, which launches benchmark subprocesses that `import torch` and initialize the Spyre runtime. PT-active utilization (`pt_util%`) is computed from a per-kernel `ideal_cycles.json` that the compiler only emits when `SENPERFORMANCE=2` is present in the environment **before** that init happens. Without it the file is not written and `pt_util%` is absent or zero, so the CI report gate and `test_perf_capture.py` (which assert `pt_util% > 0`) cannot pass.

## Change

Prefix the `spyre-perf-suite` invocation in the `perf` branch of `make tests` with `SENPERFORMANCE="${SENPERFORMANCE:-2}"`:

```make
SENPERFORMANCE="$${SENPERFORMANCE:-2}" spyre-perf-suite --no-experimental --stacks torch-spyre \
    --report "$(RESULTS_DIR)/report.txt"
```

- The variable is exported into the process environment before `spyre-perf-suite` (and the benchmark subprocesses it spawns) start, so it is present ahead of every `import torch`.
- The `${VAR:-2}` default leaves an inherited value untouched, so a base image or a caller that already sets `SENPERFORMANCE` wins.
- The prefix scopes the export to this one command, so no other `make tests` target (unit, regression, integration) changes behaviour.

## Notes

- This is the Makefile-side guarantee. A companion change sets the same variable in the base image so it is inherited on every path (model suite launchers as well as the op suite), which is the more robust fix; this Makefile prefix keeps the metric working when the suite runs off a base image that does not set it (dev laptops, other images).
- Verified with `make -n tests TEST_TYPE=perf`: the recipe expands to exactly the prefixed `spyre-perf-suite` line and nothing else is affected.

### CODE REVIEW

**Comment by HarikrishnanBalagopal** [comment]
@ashokponkumar @anubhavjana @n-marion — this one is ready for review. CI is fully green (168 success / 3 skipped, 0 failures).

The change is small and scoped: on the `make tests TEST_TYPE=perf` path it prefixes the `spyre-perf-suite` invocation with `SENPERFORMANCE=$${SENPERFORMANCE:-2}` so `pt_util%` is captured (currently 0 / N/A in reports because the compiler only writes `inductor-logs/inductor-spyre/{kernel}/perf/ideal_cycles.json` when `SENPERFORMANCE=2` is set before the runtime initializes). The `${VAR:-2}` form leaves an inherited base-image value untouched, so it composes with the base-image ENV change (companion PR spyre-frameworks#1728). PTAL when you get a chance.

**Review by ashokponkumar — APPROVED** [review]
